frontend/src/components/BCDataGrid/BCColumnSetFilter.jsx

```diff
@@ -0,0 +1,137 @@
+import {
+  forwardRef,
+  useRef,
+  useImperativeHandle,
+  useState,
+  useEffect
+} from 'react'
+import { Autocomplete, TextField, Box, Checkbox } from '@mui/material'
+import CheckBoxOutlineBlankIcon from '@mui/icons-material/CheckBoxOutlineBlank'
+import CheckBoxIcon from '@mui/icons-material/CheckBox'
+
+const icon = <CheckBoxOutlineBlankIcon fontSize="small" />
+const checkedIcon = <CheckBoxIcon fontSize="small" />
+
+const BCColumnSetFilter = forwardRef((props, ref) => {
+  const { columnWidth } = props.column.colDef
+  const [options, setOptions] = useState([])
+  const { data: optionsData, isLoading: optionsIsLoading } = props.apiQuery()
+
+  const [currentValue, setCurrentValue] = useState(null)
+  const inputRef = useRef(null)
+
+  // expose AG Grid Filter Lifecycle callbacks
+  useImperativeHandle(ref, () => {
+    return {
+      onParentModelChanged(parentModel) {
+        // When the filter is empty we will receive a null value here
+        if (!parentModel) {
+          inputRef.current.value = ''
+          setCurrentValue(null)
+        } else {
+          inputRef.current.value = parentModel.filter + ''
+          setCurrentValue(parentModel.filter)
+        }
+      }
+    }
+  })
+
+  const onInputBoxChanged = (event, input) => {
+    if (event.target.value === '') {
+      // Remove the filter
+      props.parentFilterInstance((instance) => {
+        instance.onFloatingFilterChanged(null, null)
+      })
+      return
+    }
+
+    props.parentFilterInstance((instance) => {
+      if (!props.multiple) {
+        const val = input ? input.name : ''
+        setCurrentValue(val)
+        instance.onFloatingFilterChanged('equals', val)
+      } else {
+        const filterArr = input.map((item) => item.name).join(', ')
+        setCurrentValue(filterArr)
+        instance.onFloatingFilterChanged('equals', filterArr)
+      }
+    })
+  }
+
+  useEffect(() => {
+    if (!optionsData) return
+    const optionsDataCopy = optionsData.map((option) => ({
+      name: option[props.apiOptionField ? props.apiOptionField : 'name']
+    }))
+    // if already loaded then disable re-load
+    if (options.length > 0) return
+    setOptions(optionsDataCopy)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [optionsData, props.apiOptionField])
```

Comment:
how often do these dependancies change? Just want to make sure we won't be having redundant renders.

Reply:
corrected to have dependency only on optionsData, retrieved from api query.